uniffi_bindgen/src/bindings/kotlin/gen_kotlin/mod.rs

```diff
@@ -366,26 +366,34 @@ fn object_interface_name(ci: &ComponentInterface, obj: &Object) -> String {
 
 // *sigh* - same thing for a trait, which might be either Object or CallbackInterface.
 // (we should either fold it into object or kill it!)
-fn trait_interface_name(ci: &ComponentInterface, name: &str) -> Result<String> {
-    for ci_look in ci.all_component_interfaces() {
-        let (obj_name, has_callback_interface) = match ci_look.get_object_definition(name) {
-            Some(obj) => (obj.name(), obj.has_callback_interface()),
-            None => {
-                let obj_name = match ci_look.get_callback_interface_definition(name) {
-                    Some(cbi) => cbi.name(),
-                    None => continue,
-                };
-                (obj_name, true)
-            }
-        };
-        let class_name = KotlinCodeOracle.class_name(ci_look, obj_name);
-        return if has_callback_interface {
-            Ok(class_name)
-        } else {
-            Ok(format!("{class_name}Interface"))
-        };
+fn trait_interface_name(ci: &ComponentInterface, trait_ty: &Type) -> Result<String> {
+    let namespace = match &trait_ty.module_path() {
+        Some(module_path) => ci.namespace_for_module_path(module_path)?,
+        None => bail!("Invalid trait_type: {trait_ty:?}"),
+    };
+    let Some(ci_look) = ci.find_component_interface(namespace) else {
+        anyhow::bail!("no interface {}", namespace);
```

Comment:
if you have `ci.find_component_interface` take a module path, the existing `find_ci_for` in that file could be replaced with a call to that plus the unwrap(). You don't otherwise use the namespace here so would avoid the lookup. I don't really care either way, but it seems a little cleaner?